scripts/update-projects-version.py

```diff
@@ -0,0 +1,375 @@
+
+import semver
+
+OPERATOR_NAME = "pelorus-operator"
+OPERATOR_ORGANIZATION_NAME = "pelorus"
+OPERATOR_DOMAIN = "pelorus.dora-metrics.io"
+
+ROOT = Path(__file__).resolve().parent.parent
+PELORUS_CHARTS_FOLDER = ROOT / "charts/pelorus"
+PELORUS_EXPORTERS_FOLDER = PELORUS_CHARTS_FOLDER / "charts/exporters"
+PELORUS_OPERATOR_FOLDER = ROOT / "pelorus-operator"
+PATCHES_FOLDER = ROOT / "scripts/pelorus-operator-patches"
+
+YAML_TAB = "  "
+VERSION_PATTERN = r"\d+\.\d+\.\d+(-rc\.\d+)?"
+SHELL_DEPENDENCIES = ["helm", "oc", "operator-sdk"]
+CHARTS_FILES_TO_UPDATE = {
+    # Files to updated in charts folder and number of changes in each one
+    PELORUS_CHARTS_FOLDER / "Chart.yaml": 2,
+    ROOT / "charts/operators/Chart.yaml": 1,
+    PELORUS_EXPORTERS_FOLDER / "Chart.yaml": 1,
+    PELORUS_EXPORTERS_FOLDER / "templates/_deploymentconfig.yaml": 1,
+    PELORUS_EXPORTERS_FOLDER / "templates/_imagestream_from_image.yaml": 1,
+}
+
+
+def exit_error(message: str) -> None:
+    logging.error(message)
+    raise SystemExit(1)
+
+
+def path_type(file_path: str) -> Path:
+    _file_path = Path(file_path).resolve()
+    if _file_path.exists():
+        return _file_path
+    raise argparse.ArgumentTypeError(f"Folder {_file_path} does not exist.")
+
+
+def run_command(
+    command: str, directory: Optional[Path] = None, error_message: Optional[str] = None
+) -> "subprocess.CompletedProcess[str]":
+    try:
+        return subprocess.run(
+            command,
+            shell=True,
+            check=True,
+            capture_output=True,
+            encoding="utf-8",
+            cwd=directory,
+        )
+    except subprocess.CalledProcessError as error:
+        if error_message:
+            logging.error(error_message)
+        else:
+            logging.error(str(error.stderr).strip())
+            logging.error(error)
+        raise SystemExit(error.returncode) from error
+
+
+def check_prerequisites(destination: Path) -> None:
+    if not PATCHES_FOLDER.exists():
+        exit_error(f"Patches folder {PATCHES_FOLDER} does not exist")
+
+    for dependency in SHELL_DEPENDENCIES:
+        if shutil.which(dependency) is None:
+            exit_error(
+                f"{dependency} executable not found, activate the project virtual environment"
+            )
+
+    run_command(
+        "oc auth can-i '*' '*' --all-namespaces",
+        error_message=(
+            "You must be logged in to a OpenShift cluster as a user with cluster-admin "
+            "permissions to run this script. This avoids RBAC problems. More info: "
+            "https://sdk.operatorframework.io/docs/building-operators/helm/tutorial/#prerequisites"
+        ),
+    )
+    run_command(
+        "oc get crd grafanas.integreatly.org",
+        error_message="Grafana CRD not found",
+    )
+    run_command(
+        "oc get crd prometheuses.monitoring.coreos.com",
+        error_message="Prometheus CRD not found",
+    )
+
+    if destination.is_file():
```

Comment:
The checks for the destination empty/not empty should happen before running any commands that goes outside of local box, e.g. `oc auth` they are faster and quicker to respond on the error.